### ISSUE TITLE
Build failure with GCC 14.2.0 on Ubuntu 24.10 due to simdjson issue. Bundled simdjson needs update

### ISSUE THREAD

**Issue by ilyapopov** [aras-p/ClangBuildAnalyzer#98]
Cannot build ClangBuildAnalyzer with `gcc (Ubuntu 14.2.0-4ubuntu2) 14.2.0` on Ubuntu 24.10. The error message is:

```
/home/ilya/devel/external/ClangBuildAnalyzer/src/external/simdjson/simdjson.cpp:16656:49: error: inlining failed in call to ‘always_inline’ ‘simdjson::error_code simdjson::fallback::{anonymous}::stage2::json_iterator::walk_document(V&) noexcept [with bool STREAMING = false; V = simdjson::fallback::{anonymous}::stage2::tape_builder]’: target specific option mismatch
16656 | simdjson_warn_unused simdjson_inline error_code json_iterator::walk_document(V &visitor) noexcept {
      |                                                 ^~~~~~~~~~~~~
/home/ilya/devel/external/ClangBuildAnalyzer/src/external/simdjson/simdjson.cpp:17091:39: note: called from here
17091 |   return iter.walk_document<STREAMING>(builder);
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
```

Looks like the culprit is this simdjson issue: https://github.com/simdjson/simdjson/issues/2116 and https://github.com/simdjson/simdjson/issues/2175, which is supposedly fixed in simdjson 3.9.1

It looks like someone has already submitted a PR to update simdjson: https://github.com/aras-p/ClangBuildAnalyzer/pull/97

**Comment by aras-p** [comment]
Should be fixed now that #97 is merged.

**Comment by jeaye** [comment]
Please cut a new release, now that this is fixed. :bow:
